src/database.ts

```diff
@@ -249,13 +252,29 @@ export class Database<T extends Schema> {
 				);
 
 				console.log(`Data record: `, dataRecord);
-				const dataFieldValue = parseIgnoringSuffix(
+				const parsedFieldValue = parseIgnoringSuffix(
 					await this.dataFile.get(
 						dataRecord.startByteOffset,
 						dataRecord.endByteOffset
 					),
 					this.formatType
 				);
+
+				let dataFieldValue = parsedFieldValue;
+				if (query.selectFields && query.selectFields.length > 0) {
+					if (
+						typeof parsedFieldValue === "object" &&
+						parsedFieldValue !== null
+					) {
+						dataFieldValue = query.selectFields.reduce((acc, field) => {
+							if (field in parsedFieldValue) {
+								acc[field] = parsedFieldValue[field];
+							}
+							return acc;
+						}, {} as Partial<T>);
```

Comment:
Is it possible to type the result correctly? https://stackoverflow.com/a/47232883/86433